### ISSUE TITLE
Craft 5 Support

### ISSUE THREAD

**Issue by theAdhocracy** [ethercreative/tags#31]
I realise that the entrification plan for Craft means that Tags will ultimately be rolled in Entries, but until a similar field UI is created, many people are continuing to use them. For Craft 5, this means there is no longer any way to view and edit Tags, without going via another content entry.

Basically, this plugin remains very useful and it would be great to see it get bumped to Craft 5, even if no other development work is planned 🤞

**Comment by joelzerner** [comment]
Completely agree, would love to see this ported over to Craft 5. The field UI in it's simplicity and speed is top notch and perfectly handles this type of tagging functionality.

Not sure what the migration path away from tags will be if we have to switch to using Entries instead.

**Comment by rob-c-baker** [comment]
While having this in Craft 5 would be great, due to time pressures, we took the decision to bite the bullet and convert the tags we use into entries - via the "entrification" command line tool. It worked pretty well. In terms of changes to templates - there were not that many and those that were needed were generally just things like swapping in a `sectionId` instead of a tag `groupId`. There would be more changes if querying tags from PHP context as it's a different element / model to start the query.

I started here: https://craftcms.com/docs/4.x/console-commands.html#entrify-tags - the command walks you through in a non-live environment, you can then make any template / code changes necessary and roll it all into a single deployment. There is one thing that is out with the normal deployment process: The command adjusts the data schema that the CMS works with and the last thing it does is provide the command that is needed to run on live directly (and manually) after deployment - this command moves the data from the removed tag groups into the newly created entries, based on the schema changes that arrived in the environment via project config.

Our tags are now entries, they are managed by the core CMS interface and we have one less dependency.

**Comment by smockensturm** [comment]
@joelzerner 

"The field UI in it's simplicity and speed is top notch and perfectly handles this type of tagging functionality."

https://craftcms.com/docs/5.x/reference/element-types/tags.html

"A comparable tags field UI has not yet been introduced for entries. If you or your clients value this authoring experience, it is safe to continue using tags!"

We're at 5.5 now. What's the plan @brandonkelly ?

**Comment by theAdhocracy** [comment]
Just adding in here that it is _pretty simple_ to fork this plugin and get it working with Craft 5. The only thing you need to change is the following in the `composer.json`:
```
 "require": {
    "craftcms/cms": "^5.0"
  },
```

Then modify your Craft project's `composer.json` in the [way outlined in this comment](https://github.com/ethercreative/tags/issues/26#issuecomment-1151855741), or like so (replace the bit in the github URL):

```
{
  "require": {
    "ether/tags": "dev-master"
  },
  "repositories": [
    { "type": "git", "url": "https://github.com/{...YOUR REPO...}/craft-tags" }
  ]
}
```

**Comment by JayBox325** [comment]
Thanks for your solution! I tried to fork, update and install the plugin as you described @theAdhocracy but I just get this error when I try to delete a tag, so I'm guessing it's no longer working?

<img width="452" alt="Image" src="https://github.com/user-attachments/assets/3d7f8240-9a65-4565-870f-72a1b5132a60" />

I'm also a bit nervous of launching a project that will be relying on a no-longer maintained out-of-date plugin for a system that Craft themselves will be doing away with in total this year.

**Comment by theAdhocracy** [comment]
@JayBox325 that's a fair concern. I'd say if you're creating a new project, I wouldn't advise using this plugin any more; I have several legacy projects which already rely on the functionality.

For what it's worth, I am not able to replicate that error. Are you trying to delete the tag from the Tags panel itself? If so, I will say there are two "delete" options, and you need to use the second one; the first will throw an error. I am able to delete items by checking them on the tag list, then selecting that option.

**Comment by JayBox325** [comment]
Yeah, that's my assumption too. I've made some tickets for Craft that will improve the UX of the Entrification of Tags so hopefully we get some fixes there before v6.

Yeah, that error comes from using the "delete" button without marking any.

Cheers!
